SDK fails to typecheck if used in the project with "node16" module resolution

When project with "node16" module resolution strategy uses matrix-js-sdk, it will fail to typecheck due to deep import:

```
node_modules/.pnpm/matrix-js-sdk@37.11.0/node_modules/matrix-js-sdk/lib/NamespacedValue.d.ts:1:31 - error TS2307: Cannot find module 'matrix-events-sdk/lib/types' or its corresponding type declarations.
1 import { type Optional } from "matrix-events-sdk/lib/types";
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Steps to reproduce

`package.json`:
```
{
    "name": "example",
    "version": "0.1.0",
    "description": "",
    "scripts": {
        "lint:types": "tsc --noEmit"
    },
    "type": "module",
    "main": "./lib/index.js",
    "typings": "./lib/index.d.ts",
    "dependencies": {
        "matrix-js-sdk": "^37.11.0"
    },
    "devDependencies": {
        "typescript": "^5.4.2"
    }
}
```

`tsconfig.json`:
```
{
    "compilerOptions": {
        "target": "es2022",
        "experimentalDecorators": false,
        "esModuleInterop": true,
        "module": "Node16",
        "moduleResolution": "node16",
        "noEmit": true,
        "declaration": true,
        "strict": true,
        "allowImportingTsExtensions": true,
        "lib": [
            "es2024",
            "dom"
        ],
        "baseUrl": ".",
        "typeRoots": [
            "./node_modules/@types"
        ]
    },
    "include": [
        "./src/**/*.ts"
    ]
}
```

`src/index.ts`:
```
import * as sdk from "matrix-js-sdk";
```

Then running `pnpm lint:types` with Node.js v22.14.0 will emit following error:
```
> pnpm lint:types
> example@0.1.0 lint:types example
> tsc --noEmit
node_modules/.pnpm/matrix-js-sdk@37.11.0/node_modules/matrix-js-sdk/lib/NamespacedValue.d.ts:1:31 - error TS2307: Cannot find module 'matrix-events-sdk/lib/types' or its corresponding type declarations.
1 import { type Optional } from "matrix-events-sdk/lib/types";
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 1 error in node_modules/.pnpm/matrix-js-sdk@37.11.0/node_modules/matrix-js-sdk/lib/NamespacedValue.d.ts:1
 ELIFECYCLE  Command failed with exit code 2.
```

### Solution
It's easily fixed by replacing "matrix-events-sdk/lib/types" with "matrix-events-sdk".

--- a/src/NamespacedValue.ts
+++ b/src/NamespacedValue.ts
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { type Optional } from "matrix-events-sdk/lib/types";
+import { type Optional } from "matrix-events-sdk";
 
 /**
  * Represents a simple Matrix namespaced value. This will assume that if a stable prefix

